synapseclient/models/wiki.py

```diff
@@ -1 +1,1278 @@
+            "_", "%5F"
+        )
+        return attachment_file_name_reformatted
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get the markdown file handle: {self.owner_id}"
+    )
+    async def _get_markdown_file_handle(self, synapse_client: Synapse) -> "WikiPage":
+        """Get the markdown file handle from the synapse client.
+        Arguments:
+            synapse_client: The Synapse client to use for cache access.
+        Returns:
+            A WikiPage with the updated markdown file handle id.
+        """
+        if not self.markdown:
+            return self
+        else:
+            file_path = self._to_gzip_file(
+                wiki_content=self.markdown, synapse_client=synapse_client
+            )
+            try:
+                # Upload the gzipped file to get a file handle
+                async with synapse_client._get_parallel_file_transfer_semaphore(
+                    asyncio_event_loop=asyncio.get_running_loop()
+                ):
+                    file_handle = await upload_file_handle(
+                        syn=synapse_client,
+                        parent_entity_id=self.owner_id,
+                        path=file_path,
+                    )
+                    synapse_client.logger.info(
+                        f"Uploaded file handle {file_handle.get('id')} for wiki page markdown."
+                    )
+                    # Set the markdown file handle ID from the upload response
+                    self.markdown_file_handle_id = file_handle.get("id")
+            finally:
+                # delete the temp directory saving the gzipped file
+                if os.path.exists(file_path):
+                    os.remove(file_path)
+                    synapse_client.logger.debug(f"Deleted temp directory {file_path}")
+            return self
+
+    @otel_trace_method(
+        method_to_trace_name=lambda self, **kwargs: f"Get the attachment file handles for wiki page: {self.owner_id}"
+    )
+    async def _get_attachment_file_handles(self, synapse_client: Synapse) -> "WikiPage":
+        """Get the attachment file handles from the synapse client.
+        Arguments:
+            synapse_client: The Synapse client to use for cache access.
+        Returns:
+            A WikiPage with the updated attachment file handle ids.
+        """
+        if not self.attachments:
+            return self
+        else:
+
+            async def task_of_uploading_attachment(attachment: str) -> tuple[str, str]:
+                """Process a single attachment and return its file handle ID and cache directory."""
+                # if the attachment is not a gzipped file or image file, gzip it
+                if (
+                    not attachment.endswith(".gz")
+                    and not attachment.endswith(".png")
+                    and not attachment.endswith(".jpg")
+                    and not attachment.endswith(".jpeg")
```

Comment:
Is there a list of these somewhere, or is this derived from documentation on the rest-docs? I am curious if there are additional image formats to add here.